src/ResourceManager/Consumption/Commands.Consumption.Test/ScenarioTests/BudgetTests.ps1

```diff
@@ -0,0 +1,143 @@
+	$budgetSet4 = Set-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupFilter $resourceGroupName
+	Assert-NotNull $budgetSet4
+	Assert-AreEqual $budgetName $budgetSet4.Name
+	Assert-AreEqual 1 $budgetSet4.Filter.ResourceGroups.Count
+
+	Write-Debug "Remove the budget $budgetName"
+	$response = Remove-AzureRmConsumptionBudget -Name $budgetName -PassThru
+	Assert-AreEqual True $response
+}
+
+<#
+.SYNOPSIS
+Test budget at resource group level
+#>
+function Test-BudgetAtResourceGroupLevel
+{
+	# Setup
+	$budgetName = Get-BudgetName
+	$notificationKey1 = Get-NotificationKey
+	$notificationKey2 = Get-NotificationKey
+	# This resource group is created at the scope of test subscription
+	$resourceGroupName = "RGBudgets"
+
+	Write-Debug "Create a new budget $budgetName at resource group level"
+    $budgetNew = New-AzureRmConsumptionBudget -Amount 6000 -Name $budgetName -ResourceGroupName $resourceGroupName -Category Cost -StartDate 2018-06-01 -EndDate 2018-11-01 -TimeGrain Monthly
+	Assert-NotNull $budgetNew
+	Assert-AreEqual 6000 $budgetNew.Amount
+	Assert-AreEqual $budgetName $budgetNew.Name
+	Assert-AreEqual Cost $budgetNew.Category
+	Assert-AreEqual Monthly $budgetNew.TimeGrain
+
+	Write-Debug "Get the budget $budgetName"
+	$budgetGet = Get-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupName $resourceGroupName
+	Assert-NotNull $budgetGet
+	Assert-AreEqual 6000 $budgetGet.Amount
+	Assert-AreEqual $budgetName $budgetGet.Name
+	Assert-AreEqual Cost $budgetGet.Category
+	Assert-AreEqual Monthly $budgetGet.TimeGrain
+
+	Write-Debug "Update the budget $budgetName with a notification $notificationKey when cost reaches a threshold of 90 percent of amount"
+	$budgetSet1 = Set-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupName $resourceGroupName -NotificationKey $notificationKey1 -NotificationEnabled -NotificationThreshold 90 -ContactEmail johndoe@contoso.com,janesmith@contoso.com -ContactRole Owner,Reader,Contributor
+	Assert-NotNull $budgetSet1
+	Assert-AreEqual $budgetName $budgetSet1.Name
+	Assert-AreEqual 1 $budgetSet1.Notification.Count
+
+	Write-Debug "Update the budget $budgetName with a second notificaiton $notificationKey when cost reaches a threshold of 150 percent of amount"
+	$budgetSet2 = Set-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupName $resourceGroupName -NotificationKey $notificationKey2 -NotificationEnabled -NotificationThreshold 150 -ContactEmail johndoe@contoso.com,janesmith@contoso.com -ContactRole Owner,Reader,Contributor
+	Assert-NotNull $budgetSet2
+	Assert-AreEqual $budgetName $budgetSet2.Name
+	Assert-AreEqual 2 $budgetSet2.Notification.Count
+
+	Write-Debug "Remove the budget $budgetName"
+	$response = Remove-AzureRmConsumptionBudget -Name $budgetName -ResourceGroupName $resourceGroupName -PassThru
+	Assert-AreEqual True $response
+}
+
+<#
+.SYNOPSIS
+Get all budgets at subscription level
+#>
+function Test-GetBudgets
+{
+    $budgets = Get-AzureRmConsumptionBudget 
```

Comment:
Please create at least one budget before running this so that is will always pass (and then destroy it after the call.